File: Client/src/app/_models/CommentParent.ts
export interface CommentParent{
    id: string;
    noiDung: string;
    datePosted: Date;
    displayName: string;
    photoUrl: string;
    userName: string;
    questionId: string;
    childrentComments: CommentChildren[];
}

export interface CommentChildren{
    id: number;
    noiDung: string;
    datePosted: Date;
    displayName: string;
    photoUrl: string;
    userName: string;
    parentId: string;
}

File: ForumAppCore/ForumAppCore/Entities/CommentParent.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ForumAppCore.Entities
{
    public class CommentParent
    {
        [Key]
        public Guid Id { get; set; }
        public string NoiDung { get; set; }
        public DateTime DatePosted { get; set; } = DateTime.Now;
        public AppUser AppUser { get; set; }
        public Guid UserId { get; set; }

        public Question Question { get; set; }
        public Guid QuestionId { get; set; }
        public ICollection<ChildrentComment> ChildrentComments { get; set; }
    }

    public class ChildrentComment
    {
        [Key]
        public int Id { get; set; }
        public string NoiDung { get; set; }
        public DateTime DatePosted { get; set; } = DateTime.Now;
        public AppUser AppUser { get; set; }
        public Guid UserId { get; set; }
        public Guid ParentId { get; set; }
        public CommentParent Parent { get; set; }
    }
}
